pallets/parachain-staking/src/lib.rs

```diff
@@ -90,6 +94,9 @@ pub mod pallet {
 	pub type BalanceOf<T> =
 		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 
+	pub(crate) type CollatorId<T> = <T as frame_system::Config>::AccountId;
+	pub(crate) type DelegatorId<T> = <T as frame_system::Config>::AccountId;
```

Comment:
I don't think this is necessary and don't like unnecessary code changes.

`CollatorId` is more accurately `CandidateId` but I'd rather have comments than type aliases

Reply:
I'd say in general I'd consider it a good practice to have new types making a distinction between same-type args in functions and declarations. 

Maybe it's just me but it wasn't quite obvious to me, since the pallet holds multiple maps that are sometimes indexed by `candidate`, sometimes by `delegator` and sometimes both. In case of latter, I wasn't sure which key to use as `k1` and which as `k2` since both had the same type.

My personal preference would always to be more explicit while reading code to have an immediate understanding, but would gladly remove it - if it's just me.

Reply:
I agree that it makes the code more readable, but I thank it should not be done that way: 

We can make a mistake and use an alias type instead of another one, the compiler won't be able to see it. But we tend to assume that the types are good when we review because we are used to the fact that the Rust compiler checks the consistency of the types.
So if there is an error we won't see it, whereas an error in a comment we will see it more easily. So like @4meta5 I prefer comments.

Reply:
> Maybe it's just me but it wasn't quite obvious to me, since the pallet holds multiple maps that are sometimes indexed by candidate, sometimes by delegator and sometimes both. In case of latter, I wasn't sure which key to use as k1 and which as k2 since both had the same type.

We can add a comment above the map which clarifies.

Reply:
I also find it fatiguing to keep up with things like this. The code is complex enough that every little bit of readability helps.

OTOH, I'd also prefer to have it compiler-checkable. Promoting it to a `Config` type might do the trick.

This sort of thing is probably best left to an independent PR, and probably done during a time when no big staking changes are planned to avoid ugly merge conflicts. I started to do this at one point but underestimated how much work it was / never got anywhere with it. I'd be supportive of some readability / testability / QOL changes when the time is right.